### PR TITLE
Remove task defaulting

### LINKED ISSUE #39: 🐛 | Jms tasks input defaulted (should not be)
## Describe the bug - What is the current behavior?
In dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt, 
Jms tasks have defaulted values, for example: 
```
fun ChutneyStepBuilder.JmsCleanQueueTask(
    target: String,
    destination: String,
    selector: String = "",
    bodySelector: String = "",
    browserMaxDepth: Int = 0,
    timeOut: String = "",
    strategy: Strategy? = null
) {
```
But "" is different than null

## Possible fixes
Set all optional parameters to null, not to a value (check all tasks)

### PR DESCRIPTION
**Pull request by nbrouand**
Fixes #39

### CODE REVIEW

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt [hunk 1]
```diff
@@ -1427,7 +1427,7 @@ fun ChutneyStepBuilder.RadiusAuthenticateTask(
     target: String,
     userName: String,
     userPassword: String,
-    protocol: String? = "chap",
+    protocol: String? = null,
```
Comment (boddissattva) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt:1430]: Breaking change here.
We have to inform users.

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt [hunk 2]
```diff
@@ -822,7 +822,7 @@ fun ChutneyStepBuilder.SeleniumGetTask(
         type = "selenium-get",
         inputs = listOf(
             "web-driver" to webDriver,
-            ("selector" to "tab").takeIf { newTab },
+            "selector" to newTab,
```
Comment (boddissattva) [inline dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt:825]: Breaking change here.
We have to inform users.

**Comment by boddissattva** [comment]
The method takeIfPositive() is still used ? If not, to be removed.

File: dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt (hunk 1 above)
Comment (nbrouand) [reply dsl/src/main/kotlin/com/chutneytesting/kotlin/dsl/ChutneyStepImplExtensions.kt:1430]: Already defaulted to chap in RadiusAuthenticateTask

**Review by boddissattva — APPROVED** [review]

**Pull request merged**